tools/build_pak_files.py

```diff
@@ -0,0 +1,129 @@
+    elif sys.platform == "darwin":
+        unreal_editor_bin = os.path.realpath(os.path.join(args.unreal_binaries_dir, "Mac", "UE4Editor"))
+        unreal_pak_bin    = os.path.realpath(os.path.join(args.unreal_binaries_dir, "Mac", "UnrealPak"))
+    elif sys.platform == "linux":
+        unreal_editor_bin = os.path.realpath(os.path.join(args.unreal_binaries_dir, "Linux", "UE4Editor.sh"))
+        unreal_pak_bin    = os.path.realpath(os.path.join(args.unreal_binaries_dir, "Linux", "UnrealPak.sh"))
+    
+    if args.unreal_project_dir:
+        unreal_project_dirs = [os.path.realpath(args.unreal_project_dir)]
+    else:
+        unreal_projects_dir = os.path.realpath(os.path.join(os.path.dirname(os.path.realpath(__file__)), "..", "cpp", "unreal_projects"))
+        unreal_project_dirs = [ os.path.join(unreal_projects_dir, project) for project in os.listdir(unreal_projects_dir) ]
+    
+    assert os.path.exists(args.contents_dir)
+    content_dirs = [os.path.realpath(os.path.join(args.contents_dir, x)) for x in os.listdir(args.contents_dir)]
+    assert len(content_dirs) > 0
+
+    for unreal_project_dir in unreal_project_dirs:
+
+        assert os.path.exists(unreal_project_dir)
+        _, project = os.path.split(unreal_project_dir)
+        uproject   = os.path.realpath(os.path.join(unreal_project_dir, project + ".uproject"))
+
+        for content_dir in content_dirs:
+
+            # remove existing Content dir
+            unreal_project_content_dir = os.path.join(unreal_project_dir, "Content")
+            if spear.path_exists(unreal_project_content_dir):
+                print(f"[SPEAR | build_pak_files.py] Symlink exists, removing: {unreal_project_content_dir}")
+                spear.remove_path(unreal_project_content_dir)
+
+            # create symlink
+            print(f"[SPEAR | build_pak_files.py] Creating symlink: {unreal_project_content_dir} -> {content_dir}")
+            os.symlink(content_dir, unreal_project_content_dir)
+
+            for platform in args.platforms:
+
+                cmd = [
+                    unreal_editor_bin,
+                    uproject,
+                    "-run=Cook",
+                    "-TargetPlatform=" + platform + "NoEditor",
+                    "-fileopenlog",
+                    "-ddc=InstalledDerivedDataBackendGraph",
+                    "-unversioned",
+                    "-stdout",
+                    "-FullStdOutLogOutput",
+                    "-CrashForUAT",
+                    "-unattended",
+                    "-NoLogTimes",
+                    "-UTF8Output"
+                ]
+                print(f"[SPEAR | build_pak_files.py] Executing: {' '.join(cmd)}")
+                cmd_result = subprocess.run(cmd, check=True)
+
+                platform_dir = os.path.realpath(os.path.join(unreal_project_dir, "Saved", "Cooked", f"{platform}NoEditor"))
+                content_dirs = [
```

Comment:
This inner `content_dirs` variable masks an existing `content_dirs` variable. Is this intended?

Reply:
Updated code to avoid this conflict.